sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/FieldBuilder.java

```diff
@@ -0,0 +1,256 @@
+        return searchField;
+    }
+
+    private static Field enrichWithAnnotation(Field searchField, java.lang.reflect.Field classField,
+        ClientLogger logger) {
+        if (classField.isAnnotationPresent(SimpleFieldProperty.class)
+            && classField.isAnnotationPresent(SearchableFieldProperty.class)) {
+            throw logger.logExceptionAsError(new IllegalArgumentException(
+                    String.format("@SimpleFieldProperty and @SearchableFieldProperty cannot be present simultaneously "
+                        + "for %s", classField.getName())));
+        }
+        if (classField.isAnnotationPresent(SimpleFieldProperty.class)) {
+            SimpleFieldProperty simpleFieldPropertyAnnotation =
+                classField.getDeclaredAnnotation(SimpleFieldProperty.class);
+            searchField.setSearchable(false)
+                .setSortable(simpleFieldPropertyAnnotation.isSortable())
+                .setFilterable(simpleFieldPropertyAnnotation.isFilterable())
+                .setFacetable(simpleFieldPropertyAnnotation.isFacetable())
+                .setKey(simpleFieldPropertyAnnotation.isKey())
+                .setHidden(simpleFieldPropertyAnnotation.isHidden());
+        } else if (classField.isAnnotationPresent(SearchableFieldProperty.class)) {
+            if (!searchField.getType().equals(DataType.EDM_STRING)
+                && !searchField.getType().equals(DataType.collection(DataType.EDM_STRING))) {
+                throw logger.logExceptionAsError(new RuntimeException(String.format("SearchFieldProperty can only"
+                    + " be used on string properties. Property %s returns a %s value.",
+                    classField.getName(), searchField.getType())));
+            }
+            SearchableFieldProperty searchableFieldPropertyAnnotation =
+                classField.getDeclaredAnnotation(SearchableFieldProperty.class);
+            searchField.setSearchable(true)
+                .setSortable(searchableFieldPropertyAnnotation.isSortable())
+                .setFilterable(searchableFieldPropertyAnnotation.isFilterable())
+                .setFacetable(searchableFieldPropertyAnnotation.isFacetable())
+                .setKey(searchableFieldPropertyAnnotation.isKey())
+                .setHidden(searchableFieldPropertyAnnotation.isHidden());
+            if (!searchableFieldPropertyAnnotation.analyzer().isEmpty()) {
+                searchField.setAnalyzer(AnalyzerName.fromString((searchableFieldPropertyAnnotation.analyzer())));
+            }
+            if (!searchableFieldPropertyAnnotation.searchAnalyzer().isEmpty()) {
+                searchField.setAnalyzer(AnalyzerName.fromString((searchableFieldPropertyAnnotation.searchAnalyzer())));
+            }
+            if (!searchableFieldPropertyAnnotation.indexAnalyzer().isEmpty()) {
+                searchField.setAnalyzer(AnalyzerName.fromString((searchableFieldPropertyAnnotation.indexAnalyzer())));
+            }
```

Comment:
After this, you should assert that only analyzer or both searchAnalyzer and indexAnalyzer are specified.

Reply:
Several ways to achieve this:
1. Throw error if they both categories exist. 
1. Default to analyzer when both categories exist.

Reply:
To make it simple, I will throw error for now. Can come back to this we want different behavior.